script.js

```diff
@@ -1,4 +1,218 @@
+// for player two
+var diceRollOnePlayerTwo = "";
+var diceRollTwoPlayerTwo = "";
+var playerTwoNumber = "";
+
+// leaderboard function
+var playerOneWins = 0;
+var playerTwoWins = 0;
+var leaderboardText = "";
+var leaderboard = function () {
+  if (playerOneWins > playerTwoWins) {
+    leaderboardText = `Player 1 is in the lead with a score of ${playerOneWins}. Player 2 has a score of ${playerTwoWins}.`;
+  } else if (playerOneWins < playerTwoWins) {
+    leaderboardText = `Player 2 is in the lead with a score of ${playerTwoWins}. Player 1 has a score of ${playerOneWins}.`;
+  } else {
+    leaderboardText = `Both players are currently tied with a score of ${playerTwoWins}. Let's keep playing!`;
+  }
+  return leaderboardText + "<br>" + "Hit Submit again to keep playing!";
+};
+
+// dice roll function
+var diceRoll = function () {
+  diceOutput = Math.floor(Math.random() * 6) + 1;
+  return diceOutput;
+};
+
+// player one dice roll function
+var playerOneDiceRoll = function () {
+  diceRollOnePlayerOne = diceRoll();
+  diceRollTwoPlayerOne = diceRoll();
+  if (diceRollOnePlayerOne === diceRollTwoPlayerOne) {
+    outputValue =
+      // if both dice value are the same (player skips the ordering)
+      `🎲 Welcome Player 1 🎲` +
+      "<br>" +
+      ` You rolled ${diceRollOnePlayerOne} on both dice! Your number is ${diceRollOnePlayerOne}${diceRollOnePlayerOne}. Time for Player 2 to roll!`;
+    playerOneNumber =
+      String(diceRollOnePlayerOne) + String(diceRollOnePlayerOne);
+    gameMode = gameStateArray[2];
+  } else {
+    // send player 1 to order their dice
+    outputValue =
+      `🎲 Welcome Player 1 🎲` +
+      "<br>" +
+      ` You rolled ${diceRollOnePlayerOne} on your first roll and ${diceRollTwoPlayerOne} on your second roll. Please select which die you would like to place in front by inputting '1' or '2'.`;
+    gameMode = gameStateArray[1];
+  }
+  return outputValue;
+};
+
+// player one order dice function (if both rolls are not the same)
+var playerOneDiceOrder = function (userOrder) {
+  // player 1 order dice
+  // input validation to check if number and then if it's 1 or 2
+  if (typeof userOrder === "number") {
+    if (userOrder === 1) {
+      // player 1 puts first dice roll in front
+      playerOneNumber =
+        String(diceRollOnePlayerOne) + String(diceRollTwoPlayerOne);
+      gameMode = gameStateArray[2];
+      outputValue =
+        `Your number is ${playerOneNumber}.` +
+        "<br>" +
+        ` Time for Player 2 to roll!`;
+    } else if (userOrder === 2) {
+      // player 1 puts second dice roll in front
+      playerOneNumber =
+        String(diceRollTwoPlayerOne) + String(diceRollOnePlayerOne);
+      gameMode = gameStateArray[2];
+      outputValue =
+        `Your number is ${playerOneNumber}.` +
+        "<br>" +
+        ` Time for Player 2 to roll!`;
+    } else {
+      outputValue = "Please enter either 1 or 2.";
+    }
+  } else {
+    outputValue = "Please enter either 1 or 2.";
```

Comment:
i would put the validation first in the front, probably in a separate if block just after line 68 , for a few reasons
1. Validating first is good to ensure that no rubbish input gets into your code which will reduce the likelihood of bugs
2. less nesting to make the code more readable 

another point would be that since you are outputting `Please enter either 1 or 2.` only, the user cannot see what their dice is. This is a little more on the UX but attention to such details will make your code/project/product seem that much more polished~ it will come with more coding experience~